File: generics/Demo3.java
import java.util.Arrays;
import java.util.List;

class Demo3 {
    public static void main(String[] args) {
        List<Integer> list1 = Arrays.asList(1, 2, 3, 4);
        printList(list1);

        List<Number> list2 = Arrays.asList(1, 2, 3, 4);
        printList(list2);
    }

    public static void printList(List<? super Integer> list) {
        System.out.println(list);
    }
}

File: generics/Demo3.kt
fun main() {
    val list1: List<Int> = listOf(1, 2, 3, 4)
    printList(list1)

    val list2: List<Number> = listOf(1, 2, 3, 4)
    printList(list2)
}

fun printList(list: List<in Int>) {
    println(list)
}
